src/Compilers/CSharp/Portable/Parser/LanguageParser_Patterns.cs

```diff
@@ -207,6 +207,10 @@ private PatternSyntax ParsePrimaryPattern(Precedence precedence, bool afterIs, b
 
             switch (CurrentToken.Kind)
             {
+                case SyntaxKind.DotDotToken:
+                    return _syntaxFactory.SlicePattern(
+                        CheckFeatureAvailability(EatToken(), MessageID.IDS_FeatureSlicePattern),
+                        IsPossibleSubpatternElement() ? ParseNegatedPattern(precedence, afterIs: false, whenIsKeyword: false) : null);
```

Comment:
I think `whenIsKeyword` should just be passed through. This should be observable.
I'm not sure about `afterIs` (needs further confirmation).

Reply:
In a valid code those could never happen. But it's probably a good idea to pass those args for better error recovery.